core/src/main/java/org/polypheny/db/PolyImplementation.java

```diff
@@ -284,12 +277,12 @@ public int getRowsChanged( Statement statement ) throws Exception {
             int rowsChanged;
             try {
                 Iterator<?> iterator = enumerable( statement.getDataContext() ).iterator();
-                rowsChanged = getRowsChanged( statement, iterator, getKind().name() );
+                rowsChanged = getRowsChanged( statement, iterator, MonitoringType.from( getKind() ) );
             } catch ( RuntimeException e ) {
                 if ( e.getCause() != null ) {
-                    throw new Exception( e.getCause().getMessage(), e );
+                    throw new GenericRuntimeException( e.getCause().getMessage(), e );
                 } else {
-                    throw new Exception( e.getMessage(), e );
+                    throw new GenericRuntimeException( e.getMessage(), e );
                 }
             }
             return rowsChanged;
```

Comment:
This class is declared to throw Exception. Why not also replace the exception thrown two lines below (can't comment there) with one of our new Exceptions? Then there is no longer a need to declare the class with throws Exception.

Reply:
changed